feat(frontend): Added responsive profile page layout

rohansen856 opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #81.
2. This PR does the following: Created a responsive profile page layout in /profile route according to issue #81 

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
- [x] The PR is made from a branch that's **not** called "main/master".

## Proof that changes are correct
![Screenshot 2024-06-05 165435](https://github.com/bsoc-bitbyte/busify/assets/108976786/966e6401-c236-4396-b7d0-5d3fd543f083)
![Screenshot 2024-06-05 165451](https://github.com/bsoc-bitbyte/busify/assets/108976786/3b5d73c6-4e87-4e58-93c3-5bdd6a04d7ce)



<!--
Add videos/screenshots of the user-facing interface to demonstrate that the changes
made in this PR work correctly. If this PR is for a developer-facing feature,
provide the videos/screenshots of developer-facing interface.

Please also include videos/screenshots of the developer tools
browser console, so that we can be sure that there are no console errors.

If the changes in your PRs are autogenerated via a script and you cannot provide proof
for the changes then please leave a comment "No proof of changes needed because {{Reason}}"
and remove all the sections below.
-->


## PR Pointers

- If you need a review or an answer to a question, and don't have permissions to assign people, **leave a comment** like the following: "{{Question/comment}} @{{reviewer_username}} PTAL".
- Never force push. If you do, your PR will be closed.

RISHIKESHk07 commented:
@rohansen856 LGTM, one final thing make it such that you can navigate to the profile page from the clicking the user icon / button we have at top right corner and if you are up to it make a small animation on switching tabs (in this the highligthed yellow part is above the black line , fix that a bit according to the figma design), @Pratik2026  anything else ?

rohansen856 commented:
@RISHIKESHk07 of course. i will make the necessary changes and get back to you soon. as for the animation part... is it ok to be a sliding animation of the bottom yellow bar?

rohansen856 commented:
@RISHIKESHk07 is this alright:?

https://github.com/bsoc-bitbyte/busify/assets/108976786/55989fc4-ef08-499a-bdb3-e729194c1fd2

Pratik2026 commented:
@rohansen856, we are already using `@mui/icons-material` library for importing icons. Please replace all the SVG code with corresponding mui-icons. It's not a good practice to include the whole svg code in a file.

Pratik2026 commented on frontend/src/index.css:
```diff
@@ -26,3 +26,7 @@ scrollbar-thumb {
   background-color: rgba(127, 125, 125, 0.4);
   border-radius: 10px;
 }
+
+.cover {
+  object-fit: cover;
+}
```
We prefer to add only those CSS here which we want to either apply globally or have requirement in multiple places. You can add this object-fit property using `style` attribute in `<img />` itself.

Pratik2026 commented on frontend/src/pages/Profile/index.tsx:
```diff
@@ -57,75 +64,91 @@ const ProfileImage = styled(Box)`
         <Typography
           variant="h2"
           fontSize={{xs: '1rem', sm: '1.25rem', md: '1.5rem'}}
           fontWeight={500}
```
As per the design, it should be 600 or more.

Pratik2026 replied:
@rohansen856, you have not changed the fontWeight to 600 yet.

rohansen856 replied:
@Pratik2026 i will update the font wight to the correct value.

Pratik2026 commented on frontend/src/pages/Profile/index.tsx:
```diff
@@ -57,75 +64,91 @@ const ProfileImage = styled(Box)`
+              width="100%"
             />
-          </svg>
+          ) : (
```
since profile page will be wrapped by the user protected route, therefore we don't need to conditionally render the component. To avoid "user might be null" warning, use optional operator`(?)` in image src.

rohansen856 replied:
i will remove the unneccesary code in the next pr.

Pratik2026 commented on frontend/src/pages/Profile/index.tsx:
```diff
@@ -0,0 +1,226 @@
+          <Typography
+            variant="h4"
+            fontWeight={600}
+            color={theme.palette.primary.main}
+            borderBottom={currentTab === 1 ? '4px solid' : 'none'}
+            borderColor={theme.palette.primary.main}
+            paddingBottom="0.5rem"
+            onClick={() => {
+              setCurrentTab(1);
+            }}
+          >
+            Upcoming
+          </Typography>
+          <Typography
+            variant="h4"
+            fontWeight={600}
+            color={theme.palette.primary.main}
```
unselected tab should be black in color otherwise main color of palette according to the design. You can achieve this by condtionally rendering the color according to the `currentTab` state value.

rohansen856 replied:
ok.. i will fix this issue with the suggested method

Pratik2026 commented:
@rohansen856, Also please remove the non-relevant screenshots from the PR description & update them with the new ones (keep at most 2 Screenshots)

rohansen856 commented:
@Pratik2026  is this alright?
![Screenshot 2024-06-05 164612](https://github.com/bsoc-bitbyte/busify/assets/108976786/44e50083-4d58-4cae-b94a-fc7b647b0423)
removed the unnecessary null user check from the file
i have removed the style from the global file
updated the icons from material ui
applied different color to the disabled tab
i have also created a css file named animation.css in the profile page directory for the animation. If you have any other change suggestions please tell.

rohansen856 commented:
> @rohansen856, Also please remove the non-relevant screenshots from the PR description & update them with the new ones (keep at most 2 Screenshots)

I have updated the pr screenshots to the latest design

Pratik2026 commented:
> @rohansen856 LGTM, one final thing make it such that you can navigate to the profile page from the clicking the user icon / button we have at top right corner and if you are up to it make a small animation on switching tabs (in this the highligthed yellow part is above the black line , fix that a bit according to the figma design), @Pratik2026 anything else ?

@rohansen856 I think here @RISHIKESHk07 meant to say that we have to make it such that we can navigate to the profile page by clicking on the view profile option which appears when we click on top right button.

Pratik2026 commented:
Also, the changes regarding the font-weight still remain unaddressed. Please do that as well along with the above one.

rohansen856 commented:
@Pratik2026 i have updated the fontweight and the issue with redirecting in the latest comment. thank you.

Pratik2026 commented:
merged, thanks for contributing @rohansen856

rohansen856 commented:
@Pratik2026 the pleasure is all mine!
